### ISSUE TITLE
reproduce the result on Mnist N-body

### ISSUE THREAD

**Issue by lakshya-4gp** [amazon-science/earth-forecasting-transformer#44]
Thank you so much for releasing the code. I was wondering if I need to make changes to the default config in scripts/cuboid_transformer/nbody/ cfg.yaml to reproduce the results in the paper. Currently, I'm getting sub-optimal values of 
       Test metric             DataLoader 0
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
     test_mae_epoch          48.53514099121094
     test_mse_epoch          19.73045539855957
     test_ssim_epoch        0.9322687387466431

**Comment by lakshya-4gp** [comment]
I saw that for the ENSO dataset, you'd earlier recommended someone to use a batch size of 64 instead.

**Comment by gaozhihan** [comment]
Thanks for pointing it out. 
You could change the arg `rescale_01` to be `False` in [N-body MNIST dataloaders](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/src/earthformer/datasets/nbody/nbody_mnist_torch_wrap.py#L32), i.e. to use value range `[0, 255]` instead of `[0, 1]`, to reproduce the results reported in our paper.
We keep the config `rescale_01 = True` here to be consistent with all other datasets, which rescale the data to `[0, 1]`.

**Comment by lakshya-4gp** [comment]
Thanks for the reply, I'm running the exp with new values. I also noticed that the number of global vectors, again for nbody, are set to 0
```
  num_global_vectors: 0
  use_dec_self_global: false
  dec_self_update_global: true
  use_dec_cross_global: false
  use_global_vector_ffn: false
  use_global_self_attn: false
  separate_global_qkv: false
  global_dim_ratio: 1
```
Is that correct?

**Comment by lakshya-4gp** [comment]
Also, setting rescale_01=False seems to give very high values. Is there some normalization needed on these values. SSIM seems to have improved, although still lower than the one reported in the paper.

       Test metric             DataLoader 0
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
     test_mae_epoch          10840.2353515625
     test_mse_epoch             1068347.125
     test_ssim_epoch        0.9438273906707764

**Comment by gaozhihan** [comment]
> Thanks for the reply, I'm running the exp with new values. I also noticed that the number of global vectors, again for nbody, are set to 0
> 
> ```
>   num_global_vectors: 0
>   use_dec_self_global: false
>   dec_self_update_global: true
>   use_dec_cross_global: false
>   use_global_vector_ffn: false
>   use_global_self_attn: false
>   separate_global_qkv: false
>   global_dim_ratio: 1
> ```
> 
> Is that correct?

You may expect performance gain as you set the configs of global vectors the same as those in other experiments, [e.g. in SEVIR](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/scripts/cuboid_transformer/sevir/cfg_sevir.yaml#L76-L83).

**Comment by gaozhihan** [comment]
> Also, setting rescale_01=False seems to give very high values. Is there some normalization needed on these values. SSIM seems to have improved, although still lower than the one reported in the paper.
> 
> ```
>    Test metric             DataLoader 0
> ```
> 
> ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── test_mae_epoch 10840.2353515625 test_mse_epoch 1068347.125 test_ssim_epoch 0.9438273906707764

If you have checked the validation score, you will find that the training hasn't got to optimal until the end. You need to set the [`max_epoch`](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/scripts/cuboid_transformer/moving_mnist/cfg.yaml#L14) to be 200 or even larger to see the performance saturation.
As for the MSE and MAE, you need to divide them by `255^2` and `255` since the original data is rescaled from `[0, 1]` to `[0, 255]`.

**Comment by lakshya-4gp** [comment]
thanks for the help

**Comment by lakshya-4gp** [comment]
Sorry, I checked my checkpoints, it seems like early stopping stopped the model at just the 82nd epoch, i.e. the best performance was noted at the 62nd epoch only. The default seed of 0 is used only, and no global attention is used.
     test_mae_epoch          10840.2353515625 (~42.51)
     test_mse_epoch             1068347.125  (~ 16.43)
     test_ssim_epoch        0.9438273906707764
     
In the paper, the reported value seems slightly lower, mse = 15.89, mae = 41.38 for w/o global attention cases

**Comment by gaozhihan** [comment]
> Sorry, I checked my checkpoints, it seems like early stopping stopped the model at just the 82nd epoch, i.e. the best performance was noted at the 62nd epoch only. The default seed of 0 is used only, and no global attention is used. test_mae_epoch 10840.2353515625 (~42.51) test_mse_epoch 1068347.125 (~16.43) test_ssim_epoch 0.9438273906707764
> 
> In the paper, the reported value seems slightly lower, mse = 15.89, mae = 41.38 for w/o global attention cases

Since we have applied [cosine learning rate scheduler](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/scripts/cuboid_transformer/nbody/train_cuboid_nbody.py#L362), increasing the `max_epoch` is not simply equivalent to training longer. Larger `max_epoch` also implies longer warm-up and slower learning rate decay.

**Comment by sxjscience** [comment]
@lakshya-4gp Do you still have the issue?

**Comment by lakshya-4gp** [comment]
Unfortunately, i couldn't already resources to make extra runs of experiments at the moment. I believe it's fair to close this at the moment. Thanks for your help
